esphome/components/s300/s300.cpp

```diff
@@ -0,0 +1,59 @@
+#include "s300.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace s300 {
+
+static const char *const TAG = "s300";
+
+static const uint8_t S300_START_TRANSMISSION[] = {0x62};
+static const uint8_t S300_COMMAND_READ_DATA[] = {0x52};
+
+// Time after writing command before reading i2c data (minimum is 10ms per S300 I2C programming guide)
+static const uint16_t S300_I2C_RESPONSE_WAIT_MS = 10;
+
+float S300Component::read_co2_data() {
+  uint8_t i2c_out_data[7];
+  if (this->read(i2c_out_data, 7) != i2c::ERROR_OK) {
+    ESP_LOGE(TAG, "Failed to read CO2 data from S300!");
+    this->status_set_warning("couldn't read i2c command response");
```

Comment:
```suggestion
    static const char *const message = "Couldn't read CO2 data via I2C";
    ESP_LOGE(TAG, "%s", message);
    this->status_set_warning(message);
```

`message` will have same address in memory during programs runtime as its only initialized once.

`printf("%s", message);` is safer than `printf(message);`